src/main/scala/sbtsonar/SonarPlugin.scala

```diff
@@ -41,19 +41,20 @@ object SonarPlugin extends AutoPlugin {
     sonarScan := {
       implicit val logger: Logger = streams.value.log
 
-      val maybeSonarHome = Option(System.getenv("SONAR_SCANNER_HOME"))
-        .orElse(Option(System.getProperty("sonarScanner.home")))
-
-      if (maybeSonarHome.isEmpty)
-        sys.error("SONAR_SCANNER_HOME environmental variable or sonarScanner.home system property not defined.")
+      val sonarHome = sys.env.get("SONAR_SCANNER_HOME")
+        .orElse(sys.props.get("sonarScanner.home"))
+        .getOrElse(sys.error("SONAR_SCANNER_HOME environment variable or sonarScanner.home system property not defined."))
 
       // Update the external properties file if the sonarUseExternalConfig is set to true.
       if (sonarUseExternalConfig.value)
         updatePropertiesFile(baseDirectory.value, SonarExternalConfigFileName, version.value)
 
-      val args = sonarScannerArgs(sonarUseExternalConfig.value, sonarProperties.value, version.value)
+      //Allow to set sonar properties via system properties: [https://docs.sonarqube.org/display/SONAR/Analysis+Parameters]
```

Comment:
That's a nice improvement 👍 
I'm wondering whether it would be better to populate the system properties first and then merge those with `sonarProperties` - just in case if you want to overwrite any system settings on a project level.

Reply:
For my opinion, system properties is much preferable than project-files properties. In that case you can set project name or other properties for different branches without any files changing.

Reply:
Ok, that's a valid point, especially when you e.g. run an analysis on a branch and you want to overwrite some settings in your build pipeline.